timesketch/models/acl.py

```diff
@@ -157,28 +210,52 @@ def has_permission(self, user, permission):
             user has the permission or None if the user do not have the
             permission.
         """
-        return self._get_ace(user=user, permission=unicode(permission))
+        public_ace = self.is_public
+        if public_ace:
+            return public_ace
+        return self._get_ace(permission=unicode(permission), user=user)
 
-    def grant_permission(self, user, permission):
-        """Grant permission to a user with the specific permission.
+    def grant_permission(self, permission, user=None, group=None):
+        """Grant permission to a user or group  with the specific permission.
 
         Args:
-            user: A user (Instance of timesketch.models.user.User)
             permission: Permission as string (read, write or delete)
+            user: A user (Instance of timesketch.models.user.User)
+            group: A group (Instance of timesketch.models.user.Group)
         """
-        if not self._get_ace(user, permission):
+        # Grant permission to a group.
+        if group and not self._get_ace(permission, group=group):
             self.acl.append(
-                self.AccessControlEntry(
-                    user=user, permission=permission))
+                self.AccessControlEntry(permission=permission, group=group))
             db_session.commit()
+            return
 
-    def revoke_permission(self, user, permission):
+        # Grant permission to a user.
+        if not self._get_ace(permission, user=user, check_group=False):
+            self.acl.append(
+                self.AccessControlEntry(permission=permission, user=user))
+            db_session.commit()
+
+    def revoke_permission(self, permission, user=None, group=None):
         """Revoke permission to a user with the specific permission.
```

Comment:
Oh, and this also looks like it revokes more than just user permissions, but groups as well.